test/MiddlewareListenerTest.php

```diff
@@ -82,6 +88,28 @@ public function testSuccessfullyDispatchesMiddleware()
         $this->assertEquals('Test!', $return->getBody());
     }
 
+    public function testMatchedRouteParamsAreInjectedToRequestAsAttributes()
+    {
+        $matchedRouteParam = uniqid('matched param', true);
+
+        $event = $this->createMvcEvent(
+            'foo',
+            function (ServerRequestInterface $request, ResponseInterface $response) {
+                $response->getBody()->write($request->getAttribute('myParam', 'param did not exist'));
+                return $response;
+            }
+        );
+
+        $this->routeMatch->getParams()->willReturn([
+            'myParam' => $matchedRouteParam,
+        ]);
+
+        $listener = new MiddlewareListener();
+        $return   = $listener->onDispatch($event);
+        self::assertInstanceOf(Response::class, $return);
+        self::assertSame($matchedRouteParam, $return->getBody());
```

Comment:
Please use `$this->` (instead of `self::`) in 2 above assertions to keep consistency with other tests.